File: C++/number-of-good-pairs.cpp
class Solution {
public:
    int numIdenticalPairs(vector<int>& nums) {
        int sz = nums.size();
        int good = 0;
        for (int i = 0; i < sz; i++) {
            for (int j = i + 1; j < sz; j++) {
                if (nums[i] == nums[j]) {
                    good++;
                }
            }
        }
        return good;
    }
};

File: C#/number-of-good-pairs.cs
public class Solution {
    public int NumIdenticalPairs(int[] nums) {
        var count = new Dictionary<int,int>();
        foreach (int n in nums) {
            if (!count.ContainsKey(n)) {
                count[n] = 0;
            }
            count[n]++;
        }
        int result = 0;
        foreach (var (k, v) in count) {
            result += v * (v-1) / 2;
        }
        return result;
    }
}